google/cloud/storage/examples/storage_bucket_samples.cc

```diff
@@ -576,6 +576,117 @@ void RemoveBucketLabel(google::cloud::storage::Client client, int& argc,
   (std::move(client), bucket_name, label_key);
 }
 
+void GetBucketLifecycleManagement(google::cloud::storage::Client client,
+                                  int& argc, char* argv[]) {
+  if (argc != 2) {
+    throw Usage{"get-bucket-lifecycle-management <bucket-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  //! [view_bucket_lifecycle_management] [START
+  //! storage_view_bucket_lifecycle_management]
+  namespace gcs = google::cloud::storage;
+  using ::google::cloud::StatusOr;
+  [](gcs::Client client, std::string bucket_name) {
+    StatusOr<gcs::BucketMetadata> updated_metadata =
+        client.GetBucketMetadata(bucket_name);
+
+    if (!updated_metadata) {
+      throw std::runtime_error(updated_metadata.status().message());
+    }
+
+    if (updated_metadata->has_lifecycle() &&
+        not updated_metadata->lifecycle().rule.empty()) {
```

Comment:
My reasoning is for this check is that "empty lifecycle" means disabled lifecycle management. But I'm interested in hearing if there is a more straightforward way of checking.